acceptance/tests/partitions/partitions_test.go

```diff
@@ -786,170 +359,280 @@ func TestPartitionsWithMesh(t *testing.T) {
 			helpers.Cleanup(t, cfg.NoCleanupOnFailure, func() {
 				k8s.KubectlDeleteK(t, clientClusterContext.KubectlOptions(t), kustomizeDir)
 			})
+			// This section of the tests run the in-partition networking tests.
+			t.Cleanup(func() {
```

Comment:
why is it part of the cleanup?

Reply:
If I didnt not put it inside the cleanup block, the resources from this run would hang around and would could the x-partition tests to fail.

Reply:
🤔  Now it seems that most of our test logic lives in cleanup functions, so I'm wondering if we can reorder the test somehow to make it work?

Reply:
I thought the job of the cleanup functions was to invoke a cleanup when we reached the end 🤔 
I can think about this some more

Reply:
swapping this for a `t.Run()` worked and I like it.